src/entity/trade/TradeRecipeData.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\entity\trade;
+
+final class TradeRecipeData{
```

Comment:
I don't like the name of the class, it gives me to understand that it contains the data of a single recipe, or that it's for data processing, maybe something like TradeRecipeManager, TradeRecipeGroup,  or TradeRecipesHolder would be better.

Reply:
I'm still not sure about this class name. TradeRecipeManager feels like it's too generic (which people might understand it manages all trade recipes), TradeRecipeGroup doesn't seem valid since it doesn't only contain trade recipes. TradeRecipesHolder might be good but I think we still need better name than these. (or separate other values like tier, exp to other class?)